__init__.py

```diff
@@ -311,14 +312,15 @@ def _generate_image_from_art(self, art, path):
         return image
 
     def _gen_temp_file_name(self, extension):
-        return join(self.thumbdir, mktemp()) + extension
+        _, temporary_file_name = os.path.split(mktemp())
```

Comment:
What if `TMPDIR` is not `/tmp`?

`TMPDIR=/var/tmp/ python -c 'import tempfile; print(tempfile.mktemp())'`

This won't work on Windows.

Reply:
tempfile.mktemp() returns unique file path where the unique file name is concatenated at the end of the path, so whatever the mktemp returns regardless of operating system, we are just interested in the filename which is at the end of the path, that's why i used path.split to get the last unique file name

Reply:
My bad, I incorrectly remembered that `os.path.split()` splitted path components. FWIW there's `os.path.basename()` to do this.